plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -929,26 +930,32 @@ func (c *PodAffinityChecker) anyPodMatchesPodAffinityTerm(pod *v1.Pod, allPods [
 	if len(term.TopologyKey) == 0 {
 		return false, false, errors.New("Empty topologyKey is not allowed except for PreferredDuringScheduling pod anti-affinity")
 	}
-	matchingPodExists := false
+
 	namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(pod, term)
 	selector, err := metav1.LabelSelectorAsSelector(term.LabelSelector)
 	if err != nil {
 		return false, false, err
 	}
+
+	podNum := int32(1)
 	for _, existingPod := range allPods {
 		match := priorityutil.PodMatchesTermsNamespaceAndSelector(existingPod, namespaces, selector)
 		if match {
-			matchingPodExists = true
```

Comment:
I didn't read this whole discussion but I will say that unfortunately we've been using "affinity" to mean "affinity and anti-affinity" in some places, and "affinity but not anti-affinity" in others. The top-level struct is called Affinity even though it's affinity and anti-affinity. (And then, confusingly enough, we have NodeAffinity, but both PodAffinity and PodAntiAffinity). We should have called the general concept of "affinity or anti-affinity" something like "PlacementRules." Anyway, now everything is confusing and I'm not sure there's any way out since we can't change the API in a non-backward-compatible way. But, we can try to make variable names and function names and comments as clear as possible under the very unfortunate circumstances.